modules/oci-image/01_mod.mk

```diff
@@ -54,41 +54,19 @@ $(foreach build_name,$(build_names),$(eval $(call check_variables,$(build_name))
 
 ##########################################
 
+RELEASE_DRYRUN ?= false
+
 CGO_ENABLED ?= 0
 
-build_targets := $(build_names:%=$(bin_dir)/bin/%)
-run_targets := $(build_names:%=run-%)
 oci_build_targets := $(build_names:%=oci-build-%)
 oci_push_targets := $(build_names:%=oci-push-%)
 oci_maybe_push_targets := $(build_names:%=oci-maybe-push-%)
 oci_load_targets := $(build_names:%=oci-load-%)
 
 image_tool_dir := $(dir $(lastword $(MAKEFILE_LIST)))/image_tool/
 
-$(bin_dir)/bin:
-	mkdir -p $@
-
-## Build manager binary.
-## @category [shared] Build
-$(build_targets): $(bin_dir)/bin/%: FORCE | $(NEEDS_GO) $(bin_dir)/bin
-	CGO_ENABLED=$(CGO_ENABLED) \
-	$(GO) build \
-		-ldflags '$(go_$*_ldflags)' \
-		-o $@ \
-		$(go_$*_source_path)
-
-.PHONY: $(run_targets)
-ARGS ?= # default empty
-## Run a controller from your host.
-## @category [shared] Build
-$(run_targets): run-%: | $(NEEDS_GO)
-	CGO_ENABLED=$(CGO_ENABLED) \
-	$(GO) run \
-		-ldflags '$(go_$*_ldflags)' \
-		$(go_$*_source_path) $(ARGS)
-
 .PHONY: $(oci_build_targets)
-## Build the oci image.
+## Build the OCI image.
```

Comment:
comment: It'd be great if this targets build args (CGO_ENABLED, ldflags, etc) could be more closely synchronised with those used by goreleaser / `go build` - again, aiming for a single definition so we don't test and release different things.

Not sure if it's worth the effort to do that now, though.

Reply:
That is what I'm trying to do here: https://github.com/cert-manager/makefile-modules/pull/32/files/51f9b6cf6383621401c0cacc8818d28525657b80#diff-1a68d7e0cc16504904116bcd1028f40525f8d8ea25feb5ee5559d47fa765f1e8R74-R84

I basically add the CGO_ENABLED, ldflags etc. to the gorelease config yaml file before calling gorelease release.

Reply:
I think it'd be safer to also build locally using goreleaser (i.e. don't use `go build -o $@` locally, use `$(GORELEASER) build --snapshot --clean --single-target --output $@`)

If we `go build` and use that locally (or in tests, whatever) then don't use that same built binary for a release, there are a bunch of problems that could be be introduced - say if there's a problem with setting an ldflag in the goreleaser config.